src/components/Users/CreateUserForm.tsx

```diff
@@ -116,36 +112,26 @@ export default function CreateUserForm({ onSubmitSuccess }: Props) {
     }
   }, [phoneNumber, isWhatsApp, form]);
 
-  const onSubmit = async (data: UserFormValues) => {
-    try {
-      const {
-        res,
-        data: user,
-        error,
-      } = await request(UserApi.create, {
-        body: {
-          ...data,
-          // Omit c_password as it's not needed in the API
-          c_password: undefined,
-        } as unknown as UserBase,
+  const { mutate: createUser, isPending } = useMutation({
+    mutationFn: mutate(UserApi.create),
+    onSuccess: (user: UserBase) => {
+      toast.success(t("user_added_successfully"));
+      onSubmitSuccess?.(user!);
+    },
+    onError: (error) => {
+      const errors = (error.cause?.errors as any[]) || [];
+      errors.forEach((err) => {
+        const field = err.loc[0];
+        form.setError(field, { message: err.ctx.error });
       });
+    },
+  });
 
-      if (res?.ok) {
-        Notification.Success({
-          msg: t("user_added_successfully"),
-        });
-        onSubmitSuccess?.(user!);
-      } else {
-        Notification.Error({
-          msg: error?.message ?? t("user_add_error"),
-        });
-      }
-    } catch (error) {
-      Notification.Error({
-        msg: t("user_add_error"),
-      });
-    }
-  };
+  const onSubmit = (data: z.infer<typeof userFormSchema>) =>
+    createUser(
+      (({ c_password: _c, phone_number_is_whatsapp: _p, ...payload }) =>
+        payload)(data),
+    );
```

Comment:
this is not clean. let's not do weird things to achieve something. let's maintain only readable code.

let's pick things only what we need manually here. so that later if we ever change the form for some reason (lets say we added a new field that doesn't needs to be sent to the backend, this would end up sending information that backend doesn't care about).

```
createUser({
  username: data.username,
  first_name: data.first_name,
  ...  
})
```